nwbwidgets/timeseries.py

```diff
@@ -966,4 +942,485 @@ def update(
+
+            figure.add_scattergl(
+                x=moving_average_df.centered_timestamps,
+                y=moving_average_df.moving_average,
+                showlegend=False,
+                line=dict(color="black", width=4),
+                row=1,
+                col=col_index + 1,
+            )
+
+    elif facet_col is None and facet_row is not None:
+
+        row_faceting_values = df[facet_row].dropna().unique()
+        row_faceting_values.sort()
+
+        for row_index, row_face_value in enumerate(reversed(row_faceting_values)):
+            if isinstance(row_face_value, str):
+                query_string = f"{facet_row}=='{row_face_value}'"
+            else:
+                query_string = f"{facet_row}=={row_face_value}"
+
+            sub_df = df.query(query_string)
+            # Calculate moving average
+            num_trials = sub_df["trial"].unique().size
+            moving_average_window = 2 * num_trials
+
+            moving_average_df = calculate_moving_average_over_trials(sub_df, moving_average_window)
+
+            figure.add_scattergl(
+                x=moving_average_df.centered_timestamps,
+                y=moving_average_df.moving_average,
+                showlegend=False,
+                line=dict(color="black", width=4),
+                row=row_index + 1,
+                col=1,
+            )
+    else:
+        col_faceting_values = df[facet_col].dropna().unique()
+        row_faceting_values = df[facet_row].dropna().unique()
+        for row_index, row_face_value in enumerate(reversed(row_faceting_values)):
+            for col_index, col_face_value in enumerate(col_faceting_values):
+
+                if isinstance(col_face_value, str):
+                    col_query_string = f"{facet_col}=='{col_face_value}'"
+                else:
+                    col_query_string = f"{facet_col}=={col_face_value}"
+
+                if isinstance(row_face_value, str):
+                    row_query_string = f"{facet_row}=='{row_face_value}'"
+                else:
+                    row_query_string = f"{facet_row}=={row_face_value}"
+
+                query_string = row_query_string + " and " + col_query_string
+
+                sub_df = df.query(query_string)
+                # Calculate moving average
+                num_trials = sub_df["trial"].unique().size
+                moving_average_window = 2 * num_trials
+
+                moving_average_df = calculate_moving_average_over_trials(sub_df, moving_average_window)
+
+                figure.add_scattergl(
+                    x=moving_average_df.centered_timestamps,
+                    y=moving_average_df.moving_average,
+                    showlegend=False,
+                    line=dict(color="black", width=4),
+                    row=row_index + 1,
+                    col=col_index + 1,
+                )
+
+    return figure
+
+
+def build_faceting_figure(df, facet_col, facet_row, data_label="data", trial_label="trial"):
```

Comment:
This seems specific to your widget. Maybe it would be better as a static method.